plugins/generator-1.15.2/forge-1.15.2/templates/block.java.ftl

```diff
@@ -373,14 +383,20 @@ public class ${name}Block extends ${JavaModName}Elements.ModElement {
         @Override public IFluidState getFluidState(BlockState state) {
             return state.get(WATERLOGGED) ? Fluids.WATER.getStillFluidState(false) : super.getFluidState(state);
         }
-	
+		</#if>
+
+		<#if data.isWaterloggable || hasCondition(data.placingCondition)>
 		@Override public BlockState updatePostPlacement(BlockState state, Direction facing, BlockState facingState, IWorld world, BlockPos currentPos, BlockPos facingPos) {
-	        if (state.get(WATERLOGGED)) {
-		        world.getPendingFluidTicks().scheduleTick(currentPos, Fluids.WATER, Fluids.WATER.getTickRate(world));
-	        }
-	        return super.updatePostPlacement(state, facing, facingState, world, currentPos, facingPos);
-        }
-        </#if>
+			<#if data.isWaterloggable>
+			if (state.get(WATERLOGGED)) {
+				world.getPendingFluidTicks().scheduleTick(currentPos, Fluids.WATER, Fluids.WATER.getTickRate(world));
+			}
+			</#if>
+			return <#if hasCondition(data.placingCondition)>

```

Comment:
From what I've seen in the code, blocks that "stack" (cacti, chorus plant etc.) schedule a tick update, then break in the tick method, while other blocks (torches, lanterns, bush blocks, signs for some reason) return AIR here. Should I add a checkbox for instant break/schedule tick?

Reply:
Coincidentally I only checked "stackable" blocks :D Then AIR was indeed a better choice. 

Will setting AIR update neighboring blocks, so in case one makes a custom "stackable" block using this new placement condition, it would work properly and break all stacked blocks? If not, maybe consider adding the tick option, but if setting air works in this scenario too, I think it is ok as it is.

Reply:
It does, although it instantly breaks the entire pile (the same thing happens for signs, so it shouldn't be a problem). I guess the checkbox can be added in another PR

Reply:
Surely, if it breaks, just not cascadingly, I don't see much of an issue there and think the current approach is perfectly fine.